backend/envconfig/envconfig.go

```diff
@@ -0,0 +1,79 @@
+package envconfig
+
+// TODO(issue#649): move this package into app framework
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+	"strconv"
+
+	"github.com/short-d/app/fw"
+)
+
+// EnvConfig parses configuration from environmental variables.
+type EnvConfig struct {
+	environment fw.Environment
+}
+
+// ParseConfigFromEnv retrieves configurations from environmental variables and
+// parse them into the given struct.
+func (e EnvConfig) ParseConfigFromEnv(config interface{}) error {
+	configVal := reflect.ValueOf(config)
+	if configVal.Kind() != reflect.Ptr || configVal.IsNil() {
+		return errors.New("config must be a pointer")
+	}
+
+	elem := configVal.Elem()
+	if elem.Kind() != reflect.Struct {
+		return errors.New("config must be a struct")
+	}
+
+	numFields := elem.NumField()
+	configType := elem.Type()
+
+	for idx := 0; idx < numFields; idx++ {
+		field := configType.Field(idx)
+		envName, ok := field.Tag.Lookup("env")
+		if !ok {
+			continue
+		}
+		defaultVal := field.Tag.Get("default")
+		envVal := e.environment.GetEnv(envName, defaultVal)
+		err := setFieldValue(field, elem.Field(idx), envVal)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func setFieldValue(field reflect.StructField, fieldValue reflect.Value, newValue string) error {
+	kind := field.Type.Kind()
+	switch kind {
+	case reflect.String:
+		fieldValue.SetString(newValue)
+		return nil
+	case reflect.Int:
+		num, err := strconv.Atoi(newValue)
+		if err != nil {
+			return err
+		}
+		fieldValue.SetInt(int64(num))
+		return nil
+	case reflect.Bool:
+		boolean, err := strconv.ParseBool(newValue)
+		if err != nil {
+			return err
+		}
+		fieldValue.SetBool(boolean)
+		return nil
+	default:
```

Comment:
@arberiii You can add a case for time to support parsing human readable string, such as '1week', into time.

Reply:
https://github.com/short-d/short/pull/645

Reply:
@alldroll What are your thoughts?

Reply:
I think we can reflect duration in naming. The code should be as simple as possible. And `1week` is going to work only for 1 case:)

You did a great job! Let's move this to `app` repository in the next PR

Reply:
This would definitely help with PR [#645](https://github.com/short-d/short/pull/645)!

Reply:
@alldroll naming? I think we should define human readable formats for common units and create unit package to help manage that. CacheSize int ‘env: “CACHE_SIZE” unit:”storage”’  Degrees int ‘env: “DEGREES” unit:”temperature”’

Reply:
This would definitely help with PR [#645](https://github.com/short-d/short/pull/645)!